src/ChainSafe.Gaming/Web3/Evm/JsonRpc/RpcClientProvider.cs

```diff
@@ -17,26 +17,38 @@ public class RpcClientProvider : IRpcProvider, ILifecycleParticipant
         private readonly Web3Environment environment;
         private readonly ChainRegistryProvider chainRegistryProvider;
         private readonly IChainConfig chainConfig;
+        private readonly IAnalyticsClient analyticsClient;
+        private readonly IProjectConfig projectConfig;
 
         private Network.Network network;
 
         public RpcClientProvider(
             RpcClientConfig config,
             Web3Environment environment,
             ChainRegistryProvider chainRegistryProvider,
-            IChainConfig chainConfig)
+            IChainConfig chainConfig,
+            IAnalyticsClient analyticsClient,
+            IProjectConfig projectConfig)
         {
             this.chainRegistryProvider = chainRegistryProvider;
             this.environment = environment;
             this.config = config;
             this.chainConfig = chainConfig;
+            this.analyticsClient = analyticsClient;
+            this.projectConfig = projectConfig;
 
             if (string.IsNullOrEmpty(this.config.RpcNodeUrl))
             {
                 this.config.RpcNodeUrl = chainConfig.Rpc;
             }
         }
 
+        public IAnalyticsClient AnalyticsClient => analyticsClient;
```

Comment:
To get rid of those properties you can add one more required parameter to the list of ContractBuilder's dependencies. This basically means adding the IAnalyticsClient parameter to every constructor overload. You can then store it and pass it to the Contract's constructor in Build.
IAnalyticsClient should be always registered in DI by design, that's why you don't need to create a new overload for ContractBuilder and can simply add this to every existing constructor overload.

Reply:
To access IProjectConfig and IChainConfig you can reference them as dependencies in the implementation of your IAnalyticsClient.

Reply:
I Swear I tried to do that exact thing but for whatever reason my project was constatnly failing :D 
now I asked GPT to do that for me and it works like a charm.....